code/modules/mob/living/simple_animal/hostile/venus_human_trap.dm

```diff
@@ -122,7 +122,10 @@
 		for(var/obj/O in T)
 			if(O.density)
 				return
-	vines += Beam(the_target, "vine", time=INFINITY, maxdistance = vine_grab_distance, beam_type=/obj/effect/ebeam/vine)
+	
+	var/datum/beam/newVine = Beam(the_target, "vine", time=INFINITY, maxdistance = vine_grab_distance, beam_type=/obj/effect/ebeam/vine)
+	RegisterSignal(newVine, COMSIG_PARENT_QDELETING, .proc/remove_vine, newVine)
```

Comment:
Maybe something like `.proc/remove_from_vines`? Feel free to disregard the suggestion if you feel `remove_vine` is more clear.